specification/purviewdatagovernance/PurviewDataCatalog/models.tsp

```diff
@@ -0,0 +1,1371 @@
+model CatalogModelDataAssetSource {
+  @doc("The type of the data source.")
+  type?: string;
+
+  @doc("Unique identifier of the asset.")
+  assetId?: uuid;
+
+  @doc("The type of the asset.")
+  assetType?: string;
+
+  @doc(".Type properties of the asset.")
+  assetAttributes?: string[];
+
+  @doc("Fully qualified name.")
+  fqn?: string;
+
+  @doc("Name of the account.")
+  accountName?: string;
+
+  @doc("FIXME: (utcDateTime) Please double check that this is the correct type for your scenario")
+  lastRefreshedAt?: utcDateTime;
+
+  @doc("Name of the user who last refreshed the asset.")
+  lastRefreshedBy?: string;
+}
+
+@doc("Schema of the asset.")
+model CatalogModelDataAssetSchema {
+  @doc("Name of the asset.")
+  name?: string;
+
+  @doc("Description of the asset.")
+  description?: string;
+
+  @doc("Classifications of the asset.")
+  classifications?: string[];
+
+  @doc("Type of the asset.")
+  type?: string;
+}
+
+@doc("CatalogModelDataAsset.")
+model CatalogModelDataAsset {
+  @doc("Unique identifier of the asset.")
+  id?: uuid;
+
+  @doc("Name of the asset.")
+  name?: string;
+
+  @doc("System data associated with the asset.")
+  systemData?: CatalogModelSystemDataWithExpired;
+
+  @doc("Description of the asset.")
+  description?: string;
+
+  @doc("Domain of the asset.")
+  domain?: string;
+
+  @doc("Source of the asset.")
+  source?: CatalogModelDataAssetSource;
+
+  @doc("Contacts associated with the asset.")
+  contacts?: Record<CatalogModelContactsValueInner[]>;
+
+  @doc("Classifications associated with the asset.")
+  classifications?: string[];
+
+  @doc("Unique identifier of the entity.")
+  sensitivityLabel?: uuid;
+
+  @doc("Type of the asset.")
+  type?: string;
+
+  @doc("Schema of the asset.")
+  schema?: CatalogModelDataAssetSchema[];
+
+  @doc("Lineage information associated with the asset.")
+  lineage?: string[];
+
+  @doc("Data quality score of the asset.")
+  dataQualityScore?: float64;
+
+  @doc("URL to open the asset in the catalog.")
+  openInUrl?: url;
+
+  @doc("Type properties of the asset.")
+  typeProperties?: string[];
+}
+
+@doc("DataAssetDataAssetQueryRequest.")
+model DataAssetDataAssetQueryRequest {
+  @doc("Number of items to skip.")
+  skip?: int32;
+
+  @doc("Number of items to return.")
+  top?: int32;
+
+  @doc("To order the records.")
+  orderby?: CatalogApiServiceOrderBy[];
+
+  @doc("To filter by Ids.")
+  ids?: string[];
+
+  @doc("To filter by domain Ids.")
+  domainIds?: string[];
+
+  @doc("To filter by name keyword.")
+  nameKeyword?: string;
+
+  @doc("To filter by owners.")
+  owners?: string[];
+
+  @doc("To filter by data map asset Ids.")
+  dataMapAssetIds?: string[];
+}
+
+@doc("FacetValue.")
+model FacetValue {
+  @doc("The value of the facet.")
+  value: string;
+
+  @doc("The count of items in the facet.")
+  count: string;
+}
+
+@doc("FacetsResponse.")
+@pagedResult
+model FacetsResponse {
+  @doc("The list of facets.")
+  @items
+  facets: Record<FacetValue[]>;
```

Comment:
paging items MUST be JSON array

https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#collections